Src/ILGPU/Backends/Backend.cs

```diff
@@ -752,12 +750,25 @@ public abstract class Backend<TDelegate> : Backend
         /// <summary>
         /// Returns the current intrinsic provider.
         /// </summary>
-        public IntrinsicImplementationProvider<TDelegate> IntrinsicProvider { get; }
+        public IntrinsicImplementationProvider<TDelegate> IntrinsicProvider
+        {
+            get;
+            private set;
+        }
 
         #endregion
 
         #region Methods
 
+        /// <summary>
+        /// Initializes the current intrinsic provider.
+        /// </summary>
+        protected void InitIntrinsicProvider()
+        {
+            IntrinsicProvider?.Dispose();
```

Comment:
QUESTION: `InitIntrinsicProvider` is only called from the constructor. I would expect it to always be `null` at this point - is it necessary to call `Dispose` here?

Reply:
It is not necessary at the moment; however, I thought that this could avoid programming errors in the future. But I can simply remove it for now.

Reply:
No, that's fine. Happy with the line there.